pallets/liquidity-pools-gateway/src/message_processing.rs

```diff
@@ -437,27 +422,19 @@ impl<T: Config> Pallet<T> {
 				expected_proof_count,
 			);
 
-			if let Err(e) = inbound_entry.validate(&router_ids, &router_id.clone()) {
-				return (Err(e), weight.saturating_mul(count));
-			}
-
-			if let Err(e) = Self::upsert_pending_entry(message_proof, &router_id, inbound_entry) {
```

Comment:
My eyes are not bleeding anymore when looking at this, thank you <3

Reply:
BTW, liked that you move the outbound processing impl into the `MessageProcessor`, now everything is quite unified 🙌🏻